src/extra/AutoUI/utils.ts

```diff
@@ -71,3 +77,41 @@ export const autoUIGetDisabledReason = <T extends AutoUIBaseResource<T>>(
 		});
 	}
 };
+
+export const formatSorters: Dictionary<TableSortFunction<any>> = {};
+
+export const getSortingFunction = <T extends any>(
+	schemaKey: keyof T,
+	schemaValue: JSONSchema,
+): TableSortFunction<T> => {
+	if (formatSorters[schemaValue.format ?? '']) {
+		return formatSorters[schemaValue.format ?? ''];
+	}
+
+	const types = castArray(schemaValue.type);
+	if (types.includes(JsonTypes.string)) {
+		return (a: T, b: T) => {
+			const aa = (a[schemaKey] ?? '') as string;
+			const bb = (b[schemaKey] ?? '') as string;
+			if (typeof aa === 'string' && typeof bb === 'string') {
+				return aa.toLowerCase().localeCompare(bb.toLowerCase());
+			}
+			return diff(aa, bb);
```

Comment:
we cannot cast aa and bb as string and check if they are of type string right after. This sounds wrong as well. Add a check of number and boolean on the return diff otherwise return 0. Or we can add object handler in diff function.

Reply:
Another instance of pre-existing code @JSReds. I've changed the code as per your suggestion but I'm not 100% sure I did what you had in mind, could you please check it out and confirm?